sdk/azcore/internal/pollers/util.go

```diff
@@ -185,3 +185,9 @@ func ResultHelper[T any](resp *http.Response, failed bool, out *T) error {
 	}
 	return nil
 }
+
+// IsNonTerminalHTTPStatusCode returns true if the HTTP status code should be
+// considered non-terminal thus eligible for retry.
+func IsNonTerminalHTTPStatusCode(resp *http.Response) bool {
```

Comment:
I wonder if this should be expanded to include all of the status codes [here](https://github.com/Azure/azure-sdk-for-go/pull/22675) which also happens to be our default list of HTTP status codes for retry.

Reply:
Done.